pkg/virt-controller/watch/vmi.go

```diff
@@ -2276,15 +2276,23 @@ func (c *VMIController) handleDynamicInterfaceRequests(vmi *virtv1.VirtualMachin
 
 func (c *VMIController) updateInterfaceStatus(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
 	ifaceNamingScheme := namescheme.CreateNetworkNameScheme(vmi.Spec.Networks)
+	ifaceOrdinalNamingScheme := namescheme.CreateOrdinalNetworkNameScheme(vmi.Spec.Networks)
```

Comment:
@EdDev 
> I got a bit lost in the logic of detecting the legacy naming. I actually expected to see code that checks if it is legacy or not, then act based on that information. Instead, I see both name schemes generated and worked with.
> 
> In the current form, I do not see how we can easily drop the legacy backward compatibility, as it looks to be heavily embedded and invested in all areas of the code. I hope we can simplify this more.
> 

As of today, this can work at virt-controller level, note that virt-handler doesn't have enough information to use the correct name scheme in advance.
And this is what is done in the migration pod rendering and hotplug interface flows.

The one flow that doesn't work like that (check and acts..) is virt-controller VMI update status.
I think we should avoid checking Multus network-status annotation as part of this flow because it will make it depend on the annotation for any VM.
Unlink depending on the network-status annotation only where needed - for legacy VMs (similar to migration and hotplug flows).

> > virt-controller, template: Add pod network name-scheme argument
> 
> I do not see how this commit can stand alone, it has no meaning without the implementation and usage of 
the added argument. 

This is to reduce the noise in the commit that "wire things up" and actually change Kubevirt flows.
I think it's not different that introducing a component (func, class, package) in one commit and using it in another, which is a common practice in this repo.

> > virt-handler, network setup: Pick the actual pod interface by name
> 
> The changes seem to touch too much ground. Is there a problem to create everything as if it is the new scheme and only path it later when we detect it is actually the legacy one? E.g. when we first try to fetch the link, we detect it is not there, then we go to our legacy backward compatibility fixing and update the needed.

If by ground you mean code/files/package, how so?
The code does what you said, eventually, first get the link using hashed name scheme (new), if it fails, try with  ordinal name scheme (old).
We can change newPodNIC() to create the object using the new name scheme, if it fails, try to create it using the old name scheme (basicly cosmetic change).
Lemme know if this is what you had in mind.

Reply:
I think we should avoid checking Multus network-status annotation as part of this flow because it creates a dependency on the annotation for any VM, instead of depending on the annotation only when needed - for legacy VMs (similar to migration and hotplug flows).

Reply:
Right, I got it wrong, sorry for the noise.

The code is changed to check the network-status and create the naming scheme according to it.